cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -32,8 +32,10 @@ import (
 )
 
 var (
-	migConfigFlag    config.MigConfigFlag
-	kubernetes       = flag.String("kubernetes", "", "Kuberentes master location. Leave blank for default")
+	migConfigFlag                  config.MigConfigFlag
+	kubernetes                     = flag.String("kubernetes", "", "Kuberentes master location. Leave blank for default")
+	protectAgainstUnnecessaryNodes = flag.Bool("protect-against-unnecessary-nodes", true,
```

Comment:
I'd suggest renaming this flag to better describe what we actually do. Currently user would be very surprised if we suggest him to disable this flag :) 

Maybe something like `--verify-unschedulable-pods` with a good explanation of what that means and when it could be useful.